demos/shapes-custom-question-authoring-demo/src/scorer/index.js

```diff
@@ -0,0 +1,50 @@
+
+export default class Scorer {
+    constructor(question, response) {
+        this.question = question;
+        this.response = response;
+    }
+
+    isValid() {
+        const { question, response } = this;
+
+        let isValid = false;
+
+        const hasAltResponses = question.validation.alt_responses && Array.isArray(question.validation.alt_responses) && question.validation.alt_responses.length > 0;
+
+        if (!hasAltResponses) {
+            isValid = response === question.validation.valid_response.value;
+        } else {
+            if(response === question.validation.valid_response.value) {
+                isValid = true;
+            } else {
+               isValid =  question.validation.alt_responses.map(alt => alt.value).includes(response);
+            }
+        }
+
+        return isValid;
```

Comment:
This can be simplified checking question.validation .valid_response , primary and alt_response then result validation
```suggestion

        if (!question?.validation?.valid_response) {
            return false;
        }

        // Check primary valid response
        if (response === question.validation.valid_response.value) {
            return true;
        }

        // Check alternative responses
        if (question.validation.alt_responses?.length > 0) {
            return question.validation.alt_responses.some(alt => alt?.value === response);
        }

        return false;
```